pkg/sos/serviceFuncs.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sos
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/gorilla/mux"
+)
+
+func RegistersNeccesaryPatterns(router *mux.Router) {
+	router.HandleFunc("/ping", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("pong"))
+	}).Methods("GET")
+}
+
+func RegisterServiceWithSos(service string, port uint) error {
+	return registerServiceWithSos(service, port, "http://localhost:"+PortNum)
+}
+
+func registerServiceWithSos(service string, port uint, sosServerURL string) error {
+	m := RegisterReqJson{service, port}
+	b, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest("POST", sosServerURL+"/register", bytes.NewBuffer(b))
+	if err != nil {
+		return err
+	}
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	decoder := json.NewDecoder(res.Body)
+	defer res.Body.Close()
+	var retMsg struct{ Error string }
+	if err := decoder.Decode(&retMsg); err != nil {
+		return err
+	}
+
+	if retMsg != struct{ Error string }{} {
```

Comment:
This is a bit weird to read and a bit fragile should you ever extend the regMsg struct? I wonder if you'd be better off with 
if regMsg.Error != ""

?

Reply:
Fixed.

I followed the retMsg.Error != "" solution